src/eventmaster/plugins.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/subtle"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"os"
+	"time"
+)
+
+var Plugins = map[string]Plugin{
+	"github": &GitHubPlugin{},
+}
+
+type Plugin interface {
+	ParseRequest(*http.Request, []byte) (map[string]interface{}, error)
+	ParseInfo(*http.Request, map[string]interface{}) (dc string, host string, user string, targetHostSet []string, tags []string, parentEventId string, eventTime int64, data map[string]interface{})
+	ValidateAuth(*http.Request, []byte) bool
+	GetTopics() []string
+}
+
+type GitHubPlugin struct{}
+
+func (g *GitHubPlugin) ValidateAuth(r *http.Request, payload []byte) bool {
+	secret := os.Getenv("GITHUB_SECRET_TOKEN")
+	h := hmac.New(sha1.New, []byte(secret))
+	h.Write(payload)
+
+	if signatures, ok := r.Header["X-Hub-Signature"]; ok {
+		if len(signatures) > 0 {
+			sig := []byte(signatures[0][5:])
+			rawSig := h.Sum(nil)
+			expectedSig := make([]byte, hex.EncodedLen(len(rawSig)))
+			hex.Encode(expectedSig, rawSig)
+			return subtle.ConstantTimeCompare(expectedSig, sig) == 1
+		}
+	}
+	return false
+}
+
+func (g *GitHubPlugin) ParseRequest(r *http.Request, payload []byte) (map[string]interface{}, error) {
```

Comment:
The intention of a separate parse step was to have the parse plugins be separate from the handler plugins. If we are going to have a single "github" plugin, this step doesn't buy us anything.